packages/wonder-blocks-dropdown/components/dropdown.js

```diff
@@ -0,0 +1,234 @@
+            const index = selectedValues.indexOf(selectedValue);
+            const updatedSelection = [
+                ...selectedValues.slice(0, index),
+                ...selectedValues.slice(index + 1),
+            ];
+            onChange(updatedSelection);
+        } else {
+            // Item was newly selected
+            onChange([...selectedValues, selectedValue]);
+        }
+        this.handleItemSelected();
+    };
+
+    handleOpenDropdown = () => this.setState({open: true});
+
+    handleClick = (e: SyntheticEvent<>) =>
+        this.handleOpenChanged(!this.state.open, e.type === "keyup");
+
+    _getMenuItems = (): Array<DropdownItem> => {
+        const {selectedValues} = this.props;
+        return React.Children.toArray(this.props.menuItems)
+            .filter(Boolean)
+            .map((item) => {
+                const {value, disabled} = item.props;
+                const itemObject = {
+                    component: item,
+                    focusable:
+                        ActionItem.isClassOf(item) || OptionItem.isClassOf(item)
+                            ? !disabled
+                            : false,
+                    populatedProps: {},
+                };
+                if (ActionItem.isClassOf(item)) {
+                    return {
+                        ...itemObject,
+                        populatedProps: {
+                            indent: true,
+                            onClick: this.handleItemSelected,
+                        },
+                    };
+                } else if (OptionItem.isClassOf(item)) {
+                    return {
+                        ...itemObject,
+                        populatedProps: {
+                            onToggle: this.handleOptionSelected,
+                            selected: selectedValues
+                                ? selectedValues.includes(value)
+                                : false,
+                            variant: "check",
+                        },
+                    };
+                } else {
+                    return itemObject;
+                }
+            });
+    };
+
+    render() {
+        const {
+            children,
+            alignment,
+            dropdownStyle,
+            /* eslint-disable no-unused-vars */
+            onChange,
+            selectedValues,
+            "aria-disabled": ariaDisabled,
+            ...sharedProps
+        } = this.props;
+
+        const ClickableBehavior = getClickableBehavior();
+        const isOpen = this.state.open;
+        const opener = (
+            <ClickableBehavior onClick={this.handleClick}>
+                {(state, handlers) => (
+                    <DropdownAnchor
+                        anchorRef={(ref) =>
+                            //TODO: Fix this flow error
+                            //$FlowFixMe
+                            (this.openerElement = ReactDOM.findDOMNode(ref))
```

Comment:
Nit: leave a space after the `//`

I think it's possible to resolve the flow error here.  See my comment in dropdown-anchor.js.

Reply:
Can you cast this one as well just like the one in dropdown-anchor.js.